### ISSUE TITLE
After close bash, then Git for Windows 2.13.3 show MSYS2 terminal 已停止工作.

### ISSUE THREAD

**Issue by kalsolio** [git-for-windows/git#1237]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?
```
$ git --version --build-options

** 
git version 2.13.3.windows.1
built from commit: faaf2320f32f6ac52a69502c6b157b6ea4781f50
sizeof-long: 4
machine: x86_64

**
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

** Microsoft Windows [版本 10.0.15063] **
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

** 
Path Option: Cmd
SSH Option: OpenSSH
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Enabled
**
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

** insert your response here **

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

** bash **

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
** insert your commands here **
```
 - What did you expect to occur after running these commands?

** exit **

 - What actually happened instead?

** MSYS2 terminal 已停止工作 **

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

** insert URL here **

**Comment by dscho** [comment]
> 已停止工作

Bing translate says this means "Stopped working". And I cannot reproduce that error.

**Comment by kalsolio** [comment]
![111222](https://user-images.githubusercontent.com/14342288/28238529-7716c3f6-6988-11e7-9458-7a4904bcc87b.jpg) like this.

**Comment by dscho** [comment]
Sorry, with that little to go on, I cannot help.

**Comment by kalsolio** [comment]
错误应用程序名称: mintty.exe，版本: 2.7.7.0，时间戳: 0x00fdfcfd
错误模块名称: msys-2.0.dll，版本: 2008.2.0.0，时间戳: 0x00000000
异常代码: 0xc0000005
错误偏移量: 0x00000000000ab190
错误进程 ID: 0x554
错误应用程序启动时间: 0x01d2fd6d625ffd29
错误应用程序路径: C:\Program Files\Git\usr\bin\mintty.exe
错误模块路径: C:\Program Files\Git\usr\bin\msys-2.0.dll
报告 ID: 8a652464-9f2a-4b38-918a-e4c8bd7ba13e
错误程序包全名: 
错误程序包相对应用程序 ID:

**Comment by kalsolio** [comment]
错误存储段 120796199396，类型 4
事件名称: APPCRASH
响应: 不可用
Cab Id: 0

问题签名:
P1: mintty.exe
P2: 2.7.7.0
P3: 00fdfcfd
P4: msys-2.0.dll
P5: 2008.2.0.0
P6: 00000000
P7: c0000005
P8: 00000000000ab190
P9: 
P10: 

附加文件:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER7DED.tmp.dmp
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER7E5B.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER7E6A.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER7E7B.tmp.txt

可在此处获取这些文件:
C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_mintty.exe_4d31a9b9b0c6bd1b88367d714836430b88adc_72f04e43_2421a098

分析符号: 
重新检查解决方案: 0
报告 Id: 8a652464-9f2a-4b38-918a-e4c8bd7ba13e
报告状态: 268435456
哈希存储段: 96c61397b70c886e36c4c4caac5e6398

**Comment by kalsolio** [comment]
@dscho is there any other info I shoud provide~

**Comment by kalsolio** [comment]
[WERC974.tmp.txt](https://github.com/git-for-windows/git/files/1150357/WERC974.tmp.txt)


[file.zip](https://github.com/git-for-windows/git/files/1150363/file.zip)


[WERC974.tmp.txt](https://github.com/git-for-windows/git/files/1150359/WERC974.tmp.txt)

**Comment by kalsolio** [comment]
[msys-2.0.dll.tar.gz](https://github.com/git-for-windows/git/files/1150387/msys-2.0.dll.tar.gz)
after replace with this on , not CRASH now.

**Comment by kalsolio** [comment]
![qq20170715-221728](https://user-images.githubusercontent.com/14342288/28240031-fa0eb732-69ab-11e7-909e-a37fe93b48dd.png)

**Comment by forvoid** [comment]
@kalsolio 你好我把你的msys-2.0.dll替换了之后，无法使用linux命令了。是不是这就是你所想表达的解决方式哦。我把2008换成了2006

**Comment by forvoid** [comment]
你通过其他方式解决过这个问题吗，兄弟。我现在也是遇到了这个问题。出现的问题我觉得应该是卸载git后重新安装的时候 卸载的时候不彻底

**Comment by bolls** [comment]
换成2.12.*应该没问题

**Comment by BruceZhaoR** [comment]
+1 

same problem  in `2.14.1.windows.1` version when type `exit`.

2.12.2  is OK , I want to roll back to 2.12.2 .

@dscho waiting for the new version...  ⛽️

**Comment by dscho** [comment]
> @dscho waiting for the new version... :fuelpump:

As I said, with that little to go on (I do not read Chinese, nor can I reproduce the error), I can't help. So what I am waiting for is somebody who *can* reproduce to take some ownership in the issue and debug it themselves.

**Comment by phpdever** [comment]
me too，Git Bash 2.14.0和Git Bash 2.14.1

![1](https://user-images.githubusercontent.com/17875582/29342744-14caf97a-825f-11e7-8b66-117fcec673ab.jpg)

**Comment by chucklu** [comment]
Hi @dscho ,
     I have pasted the exception detail in English,hope it helps.      

- 1.picture
![default](https://user-images.githubusercontent.com/3004057/29963421-5d1b4900-8f39-11e7-84bd-59dd8c68f6c5.png)  

- 2.event log General
Faulting application name: mintty.exe, version: 2.7.7.0, time stamp: 0x00fdfcfd
Faulting module name: msys-2.0.dll, version: 2008.2.0.0, time stamp: 0x00000000
Exception code: 0xc0000005
Fault offset: 0x00000000000ab190
Faulting process id: 0x4e20
Faulting application start time: 0x01d322c3e43a6d23
Faulting application path: C:\Program Files\Git\usr\bin\mintty.exe
Faulting module path: C:\Program Files\Git\usr\bin\msys-2.0.dll
Report Id: 38059d0e-3b63-4d36-b22a-32b45dead07a
Faulting package full name: 
Faulting package-relative application ID:        
     
- 3.event log Xml
```xml
- <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
- <System>
  <Provider Name="Application Error" /> 
  <EventID Qualifiers="0">1000</EventID> 
  <Level>2</Level> 
  <Task>100</Task> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2017-09-01T09:13:46.503844700Z" /> 
  <EventRecordID>162473</EventRecordID> 
  <Channel>Application</Channel> 
  <Computer>WASYGSHA01-1020.asnet.accorservices.net</Computer> 
  <Security /> 
  </System>
- <EventData>
  <Data>mintty.exe</Data> 
  <Data>2.7.7.0</Data> 
  <Data>00fdfcfd</Data> 
  <Data>msys-2.0.dll</Data> 
  <Data>2008.2.0.0</Data> 
  <Data>00000000</Data> 
  <Data>c0000005</Data> 
  <Data>00000000000ab190</Data> 
  <Data>4e20</Data> 
  <Data>01d322c3e43a6d23</Data> 
  <Data>C:\Program Files\Git\usr\bin\mintty.exe</Data> 
  <Data>C:\Program Files\Git\usr\bin\msys-2.0.dll</Data> 
  <Data>38059d0e-3b63-4d36-b22a-32b45dead07a</Data> 
  <Data /> 
  <Data /> 
  </EventData>
  </Event>
```

**Comment by IceWee** [comment]
I met the same error tips. 
version > 2.12.* tips ...that 
version <=2.12.* works OK.

**Comment by liwenzhuo** [comment]
replace msys-2.0.dll with the one from version<=2.12.*, then no crash

**Comment by yuansu2016** [comment]
same error.and after replace msys-2.0.dll it works.

**Comment by wjjleopard** [comment]
把系统语言改成英语，这个问题就没了。

**Comment by pfdgithub** [comment]
I have the same problem

git version 2.14.1.windows.1
built from commit: 82d9b3f3b2407b52251620597d4b14933685459d
sizeof-long: 4
machine: x86_64

Microsoft Windows [版本 10.0.15063]

C:\Program Files\Git\etc\install-options.txt

Path Option: BashOnly
SSH Option: OpenSSH
CURL Option: OpenSSL
CRLF Option: CRLFCommitAsIs
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Disabled

**Comment by chucklu** [comment]
@wjjleopard language  in operation system is English,the problem still exists.

**Comment by skynetv** [comment]
This crash not happen anymore in latest version (Git for Windows 2.14.2(2))

**Comment by dscho** [comment]
> This crash not happen anymore in latest version (Git for Windows 2.14.2(2))

Really? That would be awesome, and most likely fixed by the same fix as for #1312 or #1316.

@kalsolio can you confirm that the issue is fixed for you, too?

**Comment by pfdgithub** [comment]
it's fixed for me on 2.14.2(3)

**Comment by dscho** [comment]
Okay, that's good enough for me (sadly, it happens quite frequently that original reporters cannot be reached to confirm fixes).
